### PR TITLE
[CF-889] Require additional options to process all tenants

### PR DESCRIPTION
**Pull request by ChandraAddala**
- Added option --all-tenants to process all tenants
  - Options --all-tenants and --tenants are mutually exclusive and atleast one of them have to be present.

### CODE REVIEW

File: tiamat/src/main/scala/com/rackspace/feeds/archives/Options.scala [hunk 1]
```diff
@@ -108,8 +109,12 @@ class Options {
       opt[Seq[String]]('t', "tenants") action { (x, c) =>
 
         c.copy(tenantIds = x)
-      } text ("List of tenant IDs (comma-separated).  Default is all archiving-enabled tenants.")
-      
+      } text ("List of tenant IDs (comma-separated). Mutually exclusive with option --all-tenants")
+
+      opt[Unit]("all-tenants") action { (x, c) =>
```
Comment (shintasmith) [inline tiamat/src/main/scala/com/rackspace/feeds/archives/Options.scala]: Should we have a short option for this? Like ```-a``` perhaps?

File: tiamat/src/main/scala/com/rackspace/feeds/archives/Options.scala [hunk 2]
```diff
@@ -124,6 +129,16 @@ class Options {
       case _ => throw new Exception( "Invalid ")
     }
 
+    //if both options are used
+    if (rc.isProcessAllTenants && rc.tenantIds.nonEmpty) {
+      throw new IllegalArgumentException( "Invalid command line options. --all-tenants and -tenants cannot be given at the same time")
```
Comment (shintasmith) [inline tiamat/src/main/scala/com/rackspace/feeds/archives/Options.scala]: That should be ```--tenants```, yes? Also would be nice to print the short option. For example:
```
Invalid command line options. --all-tenants (-a) and --tenants (-t) cannot be used together.
```

**Comment by shintasmith** [comment]
Rest looks good.

**Comment by ChandraAddala** [comment]
@shintasmith Submitted an update with the changes you had suggested.

**Comment by shintasmith** [comment]
lgtm

**Pull request merged**